aggregators/1.0/modules/ROOT/pages/aggregators-module-reference.adoc

```diff
@@ -182,185 +175,163 @@ image::time-based-aggregator.png[Time based Aggregator]
 </aggregators:group-based-aggregator>
 ----
 
+If the elements reach the *Max size* specified for the group, two things occur:
 
-Aggregates elements in different groups according to a group ID.
+* The elements in that group are removed from the storage. The group is marked as complete and every new element that arrives to that group will raise an exception.
+* The *Aggregation complete* route executes with the aggregated elements of that particular group.
 
-Every time a new element reaches the aggregator, an ID will be resolved. If a group with that ID already exists in the aggregator, the value will be added to that group. Otherwise, a new group with that ID will be created and the received element will be the first element in that group's aggregation.
+Every time a new element reaches the aggregator, an ID is resolved. If a group with that ID already exists in the aggregator, the value is added to that group. Otherwise, a new group with that ID is created and the received element is the first element in that group's aggregation.
 
-Some important concepts appear with the group-based aggregator:
+Additionally, if an *Aggregator listener* is registered to the aggregator scope, the listener's callback executes with the same set of elements. +
+If the *Incremental aggregation* route is not null, and the *Max size* is not reached, then the route component chain executes with all the aggregated elements, including the last element aggregated.
 
-* Group timeout: When a group has to be released because all the necessary elements for the group did not arrive within the expected time. If a group has timed out but is not yet evicted, it will reject attempts to add any new values to that group.
+Some important concepts appear with the *Group based aggregator* scope:
 
-* Group eviction: When a group is removed from the aggregator, regardless of whether it was completed or timed out. If a new element with that group's ID is received by the aggregator, the group will be created again.
+* Group timeout +
+When a group has to be released because all the necessary elements for the group did not arrive within the expected time. If a group has timed out but is not yet evicted, it will reject attempts to add any new values to that group.
```

Comment:
```suggestion
Specifies when a group must be released because all the necessary elements for the group did not arrive within the expected time. If a group has timed out but is not yet evicted, it rejects attempts to add any new values.
```